### ISSUE TITLE
Start marker is to far from a route

### ISSUE THREAD

**Issue by 0709wiwiwi** [abrensch/brouter#568]
What is wrong ?

https://youtu.be/G82JGCHp6NE

[brouter.gpx.txt](https://github.com/nrenner/brouter-web/files/11709510/brouter.gpx.txt)
https://bikerouter.de/#map=13/24.3099/43.5004/standard&amp;lonlats=43.501053,24.328795;43.490753,24.33788&amp;profile=car-eco

**Comment by devemux86** [comment]
@0709wiwiwi the link does not seem to work, better use regular "&".

**Comment by quaelnix** [comment]
The link works fine, but I don't know what the issue is:
- This works: https://brouter.de/brouter-web/#map=13/24.3099/43.5002/standard&lonlats=43.496781,24.332315;43.508218,24.321775
- This does not: https://brouter.de/brouter-web/#map=13/24.3099/43.5002/standard&lonlats=43.501697,24.327877;43.508218,24.321775

Both times the start marker is on the same OSM way: https://www.openstreetmap.org/way/31462380

**Comment by devemux86** [comment]
Another case discussed here: https://github.com/devemux86/cruiser/discussions/205

- car-fast works: [route link](https://brouter.de/brouter-web/#map=12/24.2360/43.4667/standard&lonlats=43.459525,24.278186;43.499077,24.213858&profile=car-fast)
- car-eco not work: [route link](https://brouter.de/brouter-web/#map=12/24.2360/43.4667/standard&lonlats=43.459525,24.278186;43.499077,24.213858&profile=car-eco)
  - vmax ≥ 95 km/h works

In this case @poutnikl [CarProfiles](https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles) seem to work.

**Comment by poutnikl** [comment]
> 
> In this case @poutnikl [CarProfiles](https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles) seem to work.

That is funny, as I consider them obsolete, not updating them for years, because of the internal kinematic model, developed by abrensch for his car profiles. OTOH, they can be used as a fallback in case something goes wrong in recent development.

**Comment by afischerdev** [comment]
We have two problems around this discussion - all depend on the large distances in this area.

1. based on the sample above: `car-eco not work`: [route link](https://brouter.de/brouter-web/#map=12/24.2360/43.4667/standard&lonlats=43.459525,24.278186;43.499077,24.213858&profile=car-eco)
   The routing process  doesn't complete.. It is trying to find a route again and again. At the end a runout of points. So you have the message 'from-position not mapped ...'

   `  if (inRadius && (isFinalLink || bestPath.cost + bestPath.airdistance <= maxTotalCost + 100)) {`
   at RoutingEngine [line](https://github.com/abrensch/brouter/blob/cada37b4de5553d964476f9752dc84cf6d96a5b6/brouter-core/src/main/java/btools/router/RoutingEngine.java#L1429) is the breaker. The constant of 100 is to small in this case.
   There is an other `maxTotalCost + 100` in [line ](https://github.com/abrensch/brouter/blob/cada37b4de5553d964476f9752dc84cf6d96a5b6/brouter-core/src/main/java/btools/router/RoutingEngine.java#L1331)
   I would like to set this constant to `waypointCatchingRange` (default 250)

2. The second is base on this [sample ](https://brouter.de/brouter-web/#map=15/24.3246/43.4982/standard&lonlats=43.499764,24.329343;43.507791,24.321994). A really 'from-position not mapped' problem, the point is not found.

   ![area_arabia](https://github.com/abrensch/brouter/assets/23175706/86498bd9-86cb-48e0-ab84-6c202b809353)

   The area shows the searching area for this `from` point and the way segments around. Bouter also have a look in the eight areas around. But as you can see the top left point is not in that range. So there is no way segment for this point. 
   This is a general problem with ares like this that was planned with a ruler.
   I thought there might be a way to generate pseudo points. But for the tile generation process, this means checking all distances of all points. Might be a bit time consuming.

**Comment by devemux86** [comment]
>  I would like to set this constant to `waypointCatchingRange` (default 250)

:+1:

**Comment by abrensch** [comment]
>    I thought there might be a way to generate pseudo points. But for the tile generation process, this means checking all distances of all points. Might be a bit time consuming.


At this Stage in preprocessing there's already douglas-peuker Elimination for Points that are too dense, so doing the opposite Here should be similar in Processing cost

**Comment by devemux86** [comment]
Another BRouter example reported here: https://github.com/devemux86/cruiser/discussions/227#discussioncomment-6328482
(we see several failed routes even with waypoints exactly on road network)

I can reproduce the report with a small BRouter route:
- [Working route](https://brouter.de/brouter-web/#map=12/41.1502/-4.6386/standard&lonlats=-4.67479,41.247338;-4.594346,40.982303&profile=car-fast)
- [Not working route](https://brouter.de/brouter-web/#map=12/41.1502/-4.6386/standard&lonlats=-4.67479,41.247338;-4.651852,41.153132;-4.594346,40.982303&profile=car-fast) (adding a waypoint on a long straight highway)
- I can make it working if I add in car-fast profile: `assign waypointCatchingRange = 2000`

**Comment by afischerdev** [comment]
@abrensch 
douglas-peuker routine was a great idea. I got it runnung to add some points. But still not finding the 'from' point. I will go on deeper.

**Comment by abrensch** [comment]
Ja Not so easy. It does not Help to add it as a "Transfer node". It need to become a Network node, located in the Micro tile that does Not have one yet.

**Comment by afischerdev** [comment]
@abrensch 
Yes, I think so. At the moment I'm back to standard rd5. Not all options done here.
![area2_arabia](https://github.com/abrensch/brouter/assets/23175706/ef9ab649-1b33-4601-85aa-a57894d48587)
The points showing the area BRouter is searching for. I doesn't enter a square North of from position.
I make tests with fallback to get all the 8 blue areas around.

**Comment by afischerdev** [comment]
Did only two of the ideas for a merge. 
The samples above will work with this but I'm sure we will find more/other samples.

But I'm pausing here to add this to the necessary new APK.
